src/condor_utils/dagman_utils.cpp

```diff
@@ -637,7 +554,11 @@ DagmanUtils::processDagCommands( SubmitDagDeepOptions& deepOpts, SubmitDagShallo
 						if (!info || *info == '\0') {
 							AppendError(errMsg, "Improperly-formatted file: environment variables missing after ENV SET");
 							result = false;
-						} else { deepOpts.addToEnv.push_back(info); }
+						} else {
+							std::string kv_pairs(info);
+							trim(kv_pairs);
+							options.set("AddToEnv", kv_pairs);
```

Comment:
This prevents the stacking of MATCH lines, unless options.set() is being really clever, which is a caveat which may apply to the previous comment of this nature.  (hrm.  Worth it to add append()?)

Reply:
See above.